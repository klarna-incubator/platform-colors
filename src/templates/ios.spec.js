const generate = require('./ios');
const parseColorManifest = require('../parse-color-manifest');
const lightColors = parseColorManifest(require('./fixtures/light-colors.json'));
const darkColors = parseColorManifest(require('./fixtures/dark-colors.json'));
const highContrastColors = parseColorManifest(
  require('./fixtures/high-contrast-colors.json')
);

describe('ios template', () => {
  it.each([
    ['light', lightColors],
    ['dark', darkColors],
    ['high contrast', highContrastColors],
  ])('generates %s colors', (name, colors) => {
    expect(generate(colors)).toMatchSnapshot();
  });
});
