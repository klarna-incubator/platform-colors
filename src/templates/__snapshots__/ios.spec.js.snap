// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ios template generates dark colors 1`] = `
Array [
  Array [
    "stringColor.colorset/Contents.json",
    "{
  \\"colors\\": [
    {
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"255.000\\",
          \\"green\\": \\"255.000\\",
          \\"red\\": \\"255.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    },
    {
      \\"appearances\\": [
        {
          \\"appearance\\": \\"luminosity\\",
          \\"value\\": \\"dark\\"
        }
      ],
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"0.000\\",
          \\"green\\": \\"0.000\\",
          \\"red\\": \\"0.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    }
  ],
  \\"info\\": {
    \\"author\\": \\"xcode\\",
    \\"generator\\": \\"platform-colors\\",
    \\"version\\": 1
  }
}",
  ],
  Array [
    "alphaColor.colorset/Contents.json",
    "{
  \\"colors\\": [
    {
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"0.500\\",
          \\"blue\\": \\"0.000\\",
          \\"green\\": \\"0.000\\",
          \\"red\\": \\"255.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    },
    {
      \\"appearances\\": [
        {
          \\"appearance\\": \\"luminosity\\",
          \\"value\\": \\"dark\\"
        }
      ],
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"0.500\\",
          \\"blue\\": \\"200.000\\",
          \\"green\\": \\"200.000\\",
          \\"red\\": \\"255.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    }
  ],
  \\"info\\": {
    \\"author\\": \\"xcode\\",
    \\"generator\\": \\"platform-colors\\",
    \\"version\\": 1
  }
}",
  ],
]
`;

exports[`ios template generates high contrast colors 1`] = `
Array [
  Array [
    "stringColor.colorset/Contents.json",
    "{
  \\"colors\\": [
    {
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"204.000\\",
          \\"green\\": \\"204.000\\",
          \\"red\\": \\"204.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    },
    {
      \\"appearances\\": [
        {
          \\"appearance\\": \\"luminosity\\",
          \\"value\\": \\"dark\\"
        }
      ],
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"51.000\\",
          \\"green\\": \\"51.000\\",
          \\"red\\": \\"51.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    },
    {
      \\"appearances\\": [
        {
          \\"appearance\\": \\"contrast\\",
          \\"value\\": \\"high\\"
        }
      ],
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"255.000\\",
          \\"green\\": \\"255.000\\",
          \\"red\\": \\"255.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    },
    {
      \\"appearances\\": [
        {
          \\"appearance\\": \\"luminosity\\",
          \\"value\\": \\"dark\\"
        },
        {
          \\"appearance\\": \\"contrast\\",
          \\"value\\": \\"high\\"
        }
      ],
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"0.000\\",
          \\"green\\": \\"0.000\\",
          \\"red\\": \\"0.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    }
  ],
  \\"info\\": {
    \\"author\\": \\"xcode\\",
    \\"generator\\": \\"platform-colors\\",
    \\"version\\": 1
  }
}",
  ],
]
`;

exports[`ios template generates light colors 1`] = `
Array [
  Array [
    "stringColor.colorset/Contents.json",
    "{
  \\"colors\\": [
    {
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"255.000\\",
          \\"green\\": \\"255.000\\",
          \\"red\\": \\"255.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    }
  ],
  \\"info\\": {
    \\"author\\": \\"xcode\\",
    \\"generator\\": \\"platform-colors\\",
    \\"version\\": 1
  }
}",
  ],
  Array [
    "alphaColor.colorset/Contents.json",
    "{
  \\"colors\\": [
    {
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"0.500\\",
          \\"blue\\": \\"0.000\\",
          \\"green\\": \\"0.000\\",
          \\"red\\": \\"255.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    }
  ],
  \\"info\\": {
    \\"author\\": \\"xcode\\",
    \\"generator\\": \\"platform-colors\\",
    \\"version\\": 1
  }
}",
  ],
  Array [
    "namedColor.colorset/Contents.json",
    "{
  \\"colors\\": [
    {
      \\"color\\": {
        \\"color-space\\": \\"srgb\\",
        \\"components\\": {
          \\"alpha\\": \\"1.000\\",
          \\"blue\\": \\"255.000\\",
          \\"green\\": \\"0.000\\",
          \\"red\\": \\"0.000\\"
        }
      },
      \\"idiom\\": \\"universal\\"
    }
  ],
  \\"info\\": {
    \\"author\\": \\"xcode\\",
    \\"generator\\": \\"platform-colors\\",
    \\"version\\": 1
  }
}",
  ],
]
`;
